main.go

```diff
@@ -59,13 +61,35 @@ var (
 
 	// flags for external node
 	providerURI = fs.String("provider", "", "an URI pointing at a provider")
+
+	useNimbus = fs.Bool("nimbus", false, "Use Nimbus node")
 )
 
+// XXX: Consider isolating this with compiler flag or runtime flag.
+// If that doesn't work, and it causes issues for some reason, we can isolate
+// this change into a patch and have a custom build script for Nimbus.
+// Arrange that main.main runs on main thread.
+func init() {
```

Comment:
Would prefer to handle this with build tags. I propose to use `go build -tags nimbus` so you can create a new file:
```go
# some other file
// +build nimbus

func startNimbus() {
    runtime.LockOSThread()
    statusnim.Start()
    statusnim.ListenAndPost()
    runtime.UnlockOSThread()
}

# in main.go
func main() {
    // ...

    if *useNimbus {
        startNimbus()
    }
}
```

Reply:
LockOSThread needs to be in the init function of the main namespace AFAIK. Can you conditionally compile in Go?

We can hack it with a patch file if not for now I guess. Right now I'd like to finish this up sooner rather than later so it's in trunk and can be built on later.

Reply:
Unless we can do it very simply over a specific code block vs a file I suggest we KISS and have a patch file. Generated it and reverted main.go to origin/master version